tests/third_party/cupy/math_tests/test_sumprod.py

```diff
@@ -291,16 +286,36 @@ def test_nansum_out_wrong_shape(self):
         }
     )
 )
-@pytest.mark.usefixtures("allow_fall_back_on_numpy")
-@testing.gpu
-class TestNansumNanprodAxes(unittest.TestCase):
+class TestNansumNanprodAxes:
     @testing.for_all_dtypes(no_bool=True, no_float16=True)
     @testing.numpy_cupy_allclose(rtol=1e-6)
     def test_nansum_axes(self, xp, dtype):
         a = testing.shaped_arange(self.shape, xp, dtype)
         if not issubclass(dtype, xp.integer):
             a[:, 1] = xp.nan
-        return xp.nansum(a, axis=self.axis)
+        return xp.nansum(a, **_get_dtype_kwargs(xp, dtype), axis=self.axis)
+
+
+class TestNansumNanprodHuge:
+    def _test(self, xp, nan_slice):
+        a = testing.shaped_random((2048, 1, 1024), xp, "f")
+        a[nan_slice] = xp.nan
+        a = xp.broadcast_to(a, (2048, 256, 1024))
```

Comment:
@antonwolfy 
The size used in cupy here is  `(2048, 1024, 1024)` but I changed it to `(2048, 256, 1024)` otherwise the machine goes out of memory. Should we keep it this way? or we need to handle it differently.

Reply:
I think it is OK to keep the changes as you did. It doesn't look impacting the test scenario.